pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
The convention in the API, in go-restful's swagger generator, and in the Go json parser AIUI is that omitempty implies optional. 

Once defaults are filled in, all fields with default values would appear explicitly.